pkg/controller/install/errors.go

```diff
@@ -38,11 +38,11 @@ func IsErrorUnrecoverable(err error) bool {
 	if err == nil {
 		return false
 	}
-	_, ok := unrecoverableErrors[reasonForError(err)]
+	_, ok := unrecoverableErrors[ReasonForError(err)]
 	return ok
 }
 
-func reasonForError(err error) string {
+func ReasonForError(err error) string {
```

Comment:
Just thinking "out loud" here: it might be nice to (later!) use some of [the newer error wrapping features from the standard library](https://blog.golang.org/go1.13-errors) in place of the existing {Reason, Message} that exists today.